### DISCUSSION TITLE
Dependent query returns undefined when refetching

### DISCUSSION THREAD

**Discussion by goopi** [TanStack/query#415] [General] [answered]
Hey @tannerlinsley,

I think I might have found a bug related to dependent queries and refetching (`refetchAllOnWindowFocus: true`)

Example to reproduce it:
```js
const { data: token } = useQuery(['token', userId], getToken)
const { data: user } = useQuery(token ? ['user', token] : false, getUser)
console.log({ token, user })
...
```

When refetching I expect to get the previous —current— data but instead `user` is undefined. Is that the expected behaviour?

Two things I've noticed:
- The status of the *user* query is `loading`, unlike the status of *token* query which is always `success`
- `user` is undefined but `query.state.data` has the current data

**Comment by tannerlinsley** [comment]
Dependent Queries are always in a "success" state upon initialization, so long as a falsy query key is used. Only when the query key becomes truthy does a dependent query start the `loading, success/error` phases

**Comment by tannerlinsley** [comment]
The punctuate, this is intended behavior.

**Comment by goopi** [comment]
@tannerlinsley thanks for answering! In this case, when refreshing, the query key of the dependent query is valid (truthy) and it returns `undefined`; in case that is the intended behavior, is there a way to get the current value (`query.state.data`) to avoid changes in the UI?

**Comment by tannerlinsley** [comment, marked as answer]
Hmmm. I think you'll need to make a codesandbox to better explain what's happening.

**Reply by goopi** [reply]
Found the problem while creating the codesandbox! Thanks for you help!!

**Reply by narthur** [reply]
What was the solution?
